### ISSUE TITLE
Chef 12.4.0 and 12.4.1 - ERROR: undefined method `cheffish' for nil:NilClass

### ISSUE THREAD

**Issue by b-dean** [chef/chef#3659]
I ran accross the same error as this [StackOverflow user mentioned in their question](http://stackoverflow.com/questions/31149600/undefined-method-cheffish-for-nilnilclass-when-provisioning-chef-with-vagra)

The problem is that when there is some error during the chef-solo run, it doesn't seem to matter what kind of error, that error gets masked by the following error:

```
==> default: [2015-07-10T14:31:18+00:00] ERROR: Running exception handlers
==> default: [2015-07-10T14:31:18+00:00] ERROR: Exception handlers complete
==> default: [2015-07-10T14:31:18+00:00] ERROR: undefined method `cheffish' for nil:NilClass
==> default: [2015-07-10T14:31:18+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
```

You can see the original error in the output if you set the chef-solo provisioner log level to debug.

This seems to show up in 12.4.0 and 12.4.1

I'm going to make a repo with some sample code to reproduce the error.

**Comment by lamont-granquist** [comment]
Can you get a stacktrace?

**Comment by lamont-granquist** [comment]
chef/cheffish#55

**Comment by b-dean** [comment]
@lamont-granquist no stack trace unfortunately (as I'm sure you've figured out). Assume that cheffish fix takes care of this? Will this be in 12.4.2?

**Comment by thommay** [comment]
12.5.0 or 12.4.2, whichever comes first.

**Comment by lamont-granquist** [comment]
you can also directly update the cheffish gem in the omnibus bundle to fix it (right now you'll have to build from git).

**Comment by leor491** [comment]
After changing the cheffish version in my provision block, I came across the following:
==> default: [2015-07-20T23:26:59+02:00] FATAL: Stacktrace dumped to /var/chef/cache/chef-stacktrace.out

You should be able to find your stacktrace there on the guest:


vagrant@lucid32:~$ sudo cat /var/chef/cache/chef-stacktrace.out
Generated at 2015-07-20 23:26:59 +0200
Chef::Exceptions::MetadataNotValid: Cookbook loaded at path(s) [/tmp/vagrant-chef/39dd8d0ba49f40d9f5ae1d782bf10e0d/cookbooks/apt] has invalid metadata: The `name' attribute is required in cookbook metadata
/opt/chef/embedded/apps/chef/lib/chef/cookbook/cookbook_version_loader.rb:190:in `raise_metadata_error!'
/opt/chef/embedded/apps/chef/lib/chef/cookbook/cookbook_version_loader.rb:79:in `load'
/opt/chef/embedded/apps/chef/lib/chef/cookbook_loader.rb:83:in `block in load_cookbook'
/opt/chef/embedded/apps/chef/lib/chef/cookbook_loader.rb:82:in `each'
/opt/chef/embedded/apps/chef/lib/chef/cookbook_loader.rb:82:in `load_cookbook'
/opt/chef/embedded/apps/chef/lib/chef/cookbook_loader.rb:72:in `block in load_cookbooks'
/opt/chef/embedded/apps/chef/lib/chef/cookbook_loader.rb:71:in `each'
/opt/chef/embedded/apps/chef/lib/chef/cookbook_loader.rb:71:in `load_cookbooks'
/opt/chef/embedded/apps/chef/lib/chef/policy_builder/expand_node_object.rb:71:in `setup_run_context'
/opt/chef/embedded/apps/chef/lib/chef/client.rb:256:in `setup_run_context'
/opt/chef/embedded/apps/chef/lib/chef/client.rb:454:in `run'
/opt/chef/embedded/apps/chef/lib/chef/application.rb:271:in `block in fork_chef_client'
/opt/chef/embedded/apps/chef/lib/chef/application.rb:259:in `fork'
/opt/chef/embedded/apps/chef/lib/chef/application.rb:259:in `fork_chef_client'
/opt/chef/embedded/apps/chef/lib/chef/application.rb:225:in `block in run_chef_client'
/opt/chef/embedded/apps/chef/lib/chef/local_mode.rb:39:in `with_server_connectivity'
/opt/chef/embedded/apps/chef/lib/chef/application.rb:213:in `run_chef_client'
/opt/chef/embedded/apps/chef/lib/chef/application/solo.rb:280:in `block in interval_run_chef_client'
/opt/chef/embedded/apps/chef/lib/chef/application/solo.rb:269:in `loop'
/opt/chef/embedded/apps/chef/lib/chef/application/solo.rb:269:in `interval_run_chef_client'
/opt/chef/embedded/apps/chef/lib/chef/application/solo.rb:247:in `run_application'
/opt/chef/embedded/apps/chef/lib/chef/application.rb:60:in `run'
/opt/chef/embedded/apps/chef/bin/chef-solo:25:in `<top (required)>'
/usr/bin/chef-solo:51:in `load'

**Comment by spuder** [comment]
I can also reproduce this. The problem only happens when I have the following in my client.rb

     
client.rb
```ruby
...
cookbook_path   ["#{current_dir}"]
...
```

    chef-client -z -r recipe[foo::bar] -c C:\chef\client.rb

When I comment out this line, the problems goes away. 

Chef 12.4.1
Windows Server 2012 R2

**Comment by mconigliaro** [comment]
Changing the chef-solo log level to `:debug` helped me track this down. Like @leor491, I had an old cookbook that was missing the `name` parameter in `metadata.rb`.

**Comment by vpereira** [comment]
to add the name to the metadata.rb, works for me.

**Comment by lamont-granquist** [comment]
This is fixed in the currently released version of cheffish and will be fixed in the omnibus packages when 12.5.0 ships and pulls the current cheffish
